### PR TITLE
LibWeb: Add an initial implementation of CRC2D.clip()

### PR DESCRIPTION
**Pull request by MacDue**
With this PR Google street view, starts to kinda work (though laggy):
 
[screen-capture - 2023-04-07T142817.674.webm](https://user-images.githubusercontent.com/11597044/230617038-5cbe6fc9-7f10-423c-aac2-1155d4543155.webm)

**Boring demo:**
image

With this PR you can clip the canvas to any path you like! 
The implementation probably could be better, but this is a start :^)

### CODE REVIEW

**Review by Lubrsi — CHANGES_REQUESTED** [review]
Awesome! One little thing

File: Userland/Libraries/LibWeb/HTML/Canvas/CanvasPathClipper.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        , m_canvas_clip(canvas_clip)
+    {
+    }
+
+    RefPtr<Gfx::Bitmap const> m_saved_clip_region;
+    Gfx::IntRect m_bounding_box;
+    CanvasClip const& m_canvas_clip;
+};
+
+struct ScopedCanvasPathClip {
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/Canvas/CanvasPathClipper.h:37]: Please use `class` for data structures that aren't just a bunch of values bundled together :)

**Review by awesomekling — CHANGES_REQUESTED** [review]
Strange in commit message: Floor draw origin for in fill path implementation 
Typo in commit message: This initial implementation is fairly native

**Review by linusg — COMMENTED** [review]
Typo in second commit: `UAE` -> `UAF` (?)

**Pull request merged**